pulser-core/pulser/sequence/sequence.py

```diff
@@ -1116,6 +1109,28 @@ def add(
         """
         self._validate_channel(channel, block_eom_mode=True)
         self._add(pulse, channel, protocol)
+        if (
+            self._in_ising
+            and self._slm_mask_dmm
+            and self._schedule[self._slm_mask_dmm].get_duration() != 0
```

Comment:
Doing this here will disregard EOM pulses, so it should probably be moved to inside `self._add()` instead. I realize this will mean that you will be calling `self._add()` from within itself, which can enter a recursive loop if not handled carefuly. This is another argument in favour of having a variable registering whether the first pulse has been added - you can have something like this inside `_add()`:

```python
        if (
            self._waiting_for_first_pulse
            and self._in_ising
            and self._slm_mask_dmm
            and self.declared_channels[channel].addressing == "Global"
            and self._schedule[self._slm_mask_dmm].get_duration() != 0
            and not _ChannelSchedule.is_detuned_delay(pulse)
        ):
            ...
            self._waiting_for_first_pulse = False
            self._add(...)
```
This way, this `if` will only be entered once **and** you will be skipping all the unnecessary time-consuming checks with `get_duration()`